client/src/reducers/userReducers.js

```diff
@@ -84,3 +88,33 @@ export const UserRegisterReducer = (initialState, action) => {
       return initialState;
   }
 };
+export const UserUpdateReducer = (initialState, action) => {
+  switch (action.type) {
+    case USER_UPDATE_REQUEST:
+      return {
+        ...initialState,
+        loading: true,
+      };
+
+    case USER_UPDATE_SUCCESS:
+      return {
+        ...initialState,
+        loading: false,
+        success: true,
+        userInfo: action.payload,
+      };
+
+    case USER_UPDATE_FAIL:
+      return {
+        ...initialState,
+        loading: false,
+        error: action.payload,
+      };
+
+    case USER_UPDATE_RESET:
+      return {};

```

Comment:
@Vaibhav-Kambli we need to add '...initialState' here and only make the states you want set to null, otherwise it will overwrite the initial state for cart and recipes and that will break the application in some situation.